activerecord/test/cases/persistence_test.rb

```diff
@@ -1582,6 +1582,13 @@ def test_model_with_no_auto_populated_fields_still_returns_primary_key_after_ins
     assert_not_nil record.id
     assert record.id > 0
   end if supports_insert_returning? && !current_adapter?(:SQLite3Adapter)
+
+  def test_columns_with_default_function_are_not_overridden_after_creation
```

Comment:
Just for additional context. This test will pass on `main` and will pass on this branch but it will fail if we only remove `!_read_attribute(column)` because it reveals an issue with `_returning_columns_for_insert` having excessive columns for mysql adapters